docs: add Flutter Driver example to MCP server page (#13220)
flutter/website: sites/docs/src/content/ai/mcp-server.md

[`package:fl_chart`]: {{site.pub-pkg}}/fl_chart

### Interact with a running app

You can use the Dart and Flutter MCP server to drive a running
Flutter app from your AI assistant—take screenshots, tap buttons,
enter text, scroll, and hot reload.

On mobile and desktop, gate `enableFlutterDriverExtension()` behind a
`--dart-define` flag in your app's `main()` so it stays out of
production builds:

```dart

> import 'package:flutter_driver/driver_extension.dart';

Reviewer: Since users might not have `flutter_driver` added as a dependency, should we suggest adding it first?

````md
```console
$ flutter pub add "flutter_driver:{sdk: flutter}"
```
````

Author: Good call — added a `flutter pub add "flutter_driver:{sdk: flutter}"` step before the snippet in 649bfc7. It has to be a regular dependency (not a dev_dependency) since it's imported from `main()`.

void main() {
  if (const bool.fromEnvironment('ENABLE_FLUTTER_DRIVER')) {
    enableFlutterDriverExtension();
  }
  runApp(const MyApp());
}
```

Launch your app with the flag turned on:

```bash
flutter run -d <device-id> --dart-define=ENABLE_FLUTTER_DRIVER=true

> ```

Reviewer: To match the setup we use across the site:

````suggestion
```console
$ flutter run -d <device-id> --dart-define=ENABLE_FLUTTER_DRIVER=true
```
````

Then ask your assistant to connect:

> Connect to my running Flutter app, take a screenshot, then tap "Sign In".

The agent uses the `dtd` tool to discover the app and `flutter_driver_command`

> to drive its UI—no integration test setup required.

Reviewer: Seems we can maybe simplify this?

```suggestion
The agent uses the `dtd` tool to discover the app and
`flutter_driver_command` to drive its UI.
```

:::note
**Web**: the `flutter_driver` package doesn't compile under dart2js. Pair
the Dart MCP server with a browser-driving MCP for clicks and screenshots;
the Dart MCP server still handles widget tree, runtime errors, and hot

> reload through DTD. Prefer `flutter run -d web-server` so the browser the

Reviewer: I'm a bit confused here. So does this strategy not work with `dart2js` (release mode) or web as a whole? Since it at first mentions dart2js but here mentioned hot reload, which is not available with dart2js anyway.

Author: You're right, that was contradictory. `dart2js` is the release/profile compiler, while hot reload only runs in debug mode (DDC) — so pairing the two read as nonsense. The actual point is that the `flutter_driver` extension isn't supported on *any* web build (debug or release); the DTD-based features (widget tree, runtime errors, hot reload) are a separate path that works in a normal `flutter run` web **debug** session. Reworded in 649bfc7 to drop the dart2js framing and say "not supported on web builds."

> agent drives is the one DTD is connected to—with -d chrome, only the

Reviewer: ```suggestion
agent drives is the one DTD is connected to—with `-d chrome`, only the
```

window Flutter spawned receives hot reload patches.

> See the [setup guide][flutter-driver-guide] for the snippet, web modes,

Reviewer: What snippet?

Author: Fair — "the snippet" was ambiguous since one's already shown here on the page. I meant the web-safe conditional-import + stub version that lives in the guide (different from the simplified inline one above). Reworded in 649bfc7 to name it explicitly: "the web-safe conditional import."

and common pitfalls.
:::

> [flutter-driver-guide]: https://github.com/dart-lang/ai/blob/main/pkgs/dart_mcp_server/README.md#connect-to-a-running-flutter-app

Reviewer: I guess this PR should wait to land until the one to `dart-lang/ai` does for this link?

Author: Yes — that link points to the `#connect-to-a-running-flutter-app` section that [dart-lang/ai#414](https://github.com/dart-lang/ai/pull/414) adds. That PR is still open, so the anchor 404s on `main` until it lands. This one should merge after (or together with) #414. Happy to keep it in draft / on hold until then.
